src/index.js

```diff
@@ -30,13 +30,32 @@ if (
   typeof navigator !== 'undefined' &&
   navigator.product === 'ReactNative'
 ) {
+  // eslint-disable-next-line no-console
   console.warn(
     "It looks like you've imported 'styled-components' on React Native.\n" +
       "Perhaps you're looking to import 'styled-components/native'?\n" +
       'Read more about this at https://www.styled-components.com/docs/basics#react-native'
   )
 }
 
+/* Warning if there are several instances of styled-components */
+if (process.env.NODE_ENV !== 'production' && typeof window !== 'undefined') {
```

Comment:
Any specific reason for ignoring the server side case (`typeof window !== 'undefined'`)? Was it linked to HMR issues or the lack of a good global ponyfill?
```js
// https://github.com/zloirock/core-js/issues/86#issuecomment-115759028
export default (typeof window != 'undefined' && window.Math == Math
  ? window
  : typeof self != 'undefined' && self.Math == Math
  ? self
  : Function('return this')());
```